From 4a1033664369ed00fadecb57a09dbb74db0a72b8 Mon Sep 17 00:00:00 2001
From: Wan Ahmad Zainie <wan.ahmad.zainie.wan.mohamad@intel.com>
Date: Thu, 13 Apr 2017 12:04:43 +0800
Subject: [PATCH 37/37] x86/platform/bxt: Add pin mappings for Apollo Lake

HS UART runtime PM support relies on different pin states, which must be
provided and registered. Define and register pin mappings for
HS UART (ttyS2).

These pin mappings should come from ACPI directly.

Signed-off-by: Wan Ahmad Zainie <wan.ahmad.zainie.wan.mohamad@intel.com>
---
 arch/x86/platform/bxt/apl-board.c | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/arch/x86/platform/bxt/apl-board.c b/arch/x86/platform/bxt/apl-board.c
index 7b81664..278ac22 100644
--- a/arch/x86/platform/bxt/apl-board.c
+++ b/arch/x86/platform/bxt/apl-board.c
@@ -21,6 +21,18 @@
 #include <linux/spi/spi.h>
 #include <linux/spi/pxa2xx_spi.h>
 #include <linux/pwm.h>
+#include <linux/pinctrl/machine.h>
+#include <linux/pinctrl/pinconf-generic.h>
+
+/* Ideally, pin mappings should be provided by ACPI */
+static const struct pinctrl_map uart2_mappings[] __initconst = {
+	PIN_MAP_MUX_GROUP("dw-apb-uart.10", PINCTRL_STATE_INIT, "INT3452:00",
+			  "uart2_in_gpio_grp", "uart2_gpio"),
+	PIN_MAP_MUX_GROUP("dw-apb-uart.10", PINCTRL_STATE_DEFAULT, "INT3452:00",
+			  "uart2_in_uart_grp", "uart2_uart"),
+	PIN_MAP_MUX_GROUP("dw-apb-uart.10", PINCTRL_STATE_SLEEP, "INT3452:00",
+			  "uart2_in_gpio_grp", "uart2_gpio"),
+};
 
 static struct pxa2xx_spi_chip chip_data = {
 	.gpio_cs = -EINVAL,
@@ -131,4 +143,12 @@ static int __init apl_board_init(void)
 	return ret;
 }
 arch_initcall(apl_board_init);
+
+static int __init uart2_init(void)
+{
+	return pinctrl_register_mappings(uart2_mappings,
+					 ARRAY_SIZE(uart2_mappings));
+}
+postcore_initcall(uart2_init);
+
 MODULE_LICENSE("GPL v2");
-- 
1.9.1

